### ISSUE TITLE
Guzzle 7 Support

### ISSUE THREAD

**Issue by elijahchancey** [keenlabs/KeenClient-PHP#139]
Please add support for Guzzle 7. Thanks!

**Comment by stof** [comment]
The `guzzlehttp/guzzle-services` package has not been updated yet to be compatible with `guzzlehttp/guzzle` 7, so this is impossible for now.

**Comment by elijahchancey** [comment]
The last commit to guzzle-services was Oct 6, 2017.  https://github.com/guzzle/guzzle-services

Hmmmm.

**Comment by elijahchancey** [comment]
Looks like the maintainer of guzzle-services isn't interested in maintaining it anymore. Is there any chance your project could remove its dependency on guzzle-services?

**Comment by elijahchancey** [comment]
Any update on this? We'd like to migrate to Laravel 8 (which uses guzzle 7), but we can't because keen doesn't support it yet. thanks! @stof

**Comment by stof** [comment]
@elijahchancey we're blocked by guzzle-services. The PR adding support for Guzzle 7 in it is not merged yet.

but Laravel 8 still supports Guzzle 6.5, so you could still downgrade Guzzle in the meantime.

**Comment by elijahchancey** [comment]
Laravel 8 requires Guzzle 7. 
https://laravel.com/docs/8.x/upgrade#updating-dependencies

Would it make sense for Keen to drop an abandoned dependency? @stof

**Comment by stof** [comment]
@elijahchancey there is a difference between their doc and their own testsuite then: https://github.com/laravel/framework/blob/f6783a510d1380abca0ffec2eecdc126f69dc0fe/composer.json#L84

**Comment by elijahchancey** [comment]
Thanks. I'll open a laravel issue so they can clarify.

**Comment by elijahchancey** [comment]
@stof Looks like Guzzle 7 support has been added to guzzle-services! https://github.com/guzzle/guzzle-services/blob/master/CHANGELOG.md

**Comment by elijahchancey** [comment]
@piotrek-buchman @niektenhoopen @MohammadWaleed Help?

**Comment by MohammadWaleed** [comment]
> @piotrek-buchman @niektenhoopen @MohammadWaleed Help?

Hello @elijahchancey , 
i opened a new pull request to update the versions to 7.0 and 1.2 for both Guzzle and Guzzle Services https://github.com/keenlabs/KeenClient-PHP/pull/140

**Comment by niektenhoopen** [comment]
> > @piotrek-buchman @niektenhoopen @MohammadWaleed Help?
> 
> Hello @elijahchancey ,
> i opened a new pull request to update the versions to 7.0 and 1.2 for both Guzzle and Guzzle Services #140

You are a good person 👍  I already typed a comment to ask @elijahchancey for a PR himself instead of mentioning random people but this is a better and nicer solution. Hope he checks out the changes.

**Comment by elijahchancey** [comment]
Given how much money my company spends on Keen, I'd rather the regular contributors (who are presumably employees of Keen) handle the details. Thanks for the PR Mohammad!

**Comment by stof** [comment]
@elijahchancey I'm a regular contributor on this project, but I'm not a Keen employee at all.

**Comment by niektenhoopen** [comment]
> Given how much money my company spends on Keen, I'd rather the regular contributors (who are presumably employees of Keen) handle the details. Thanks for the PR Mohammad!

Maybe check who you are mentioning first? I don't even consider myself a regular contributor. We are just spending a lot of much money on Keen (too). This is not how a community works. If you want to reach someone working at Keen, you should probably get in touch with customer support IMHO.

**Comment by MohammadWaleed** [comment]
I'm not even using Keen anymore, i just wanted to give back to the community and help where i can

**Comment by stof** [comment]
@michalpierog this issue should be closed as support for Guzzle 7 has already been added in 2.8.0

**Comment by michalpierog** [comment]
Released in v2.8.0.
